azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool.go

```diff
@@ -0,0 +1,321 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmVirtualDesktopHostPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmVirtualDesktopHostPoolCreateUpdate,
+		Read:   resourceArmVirtualDesktopHostPoolRead,
+		Update: resourceArmVirtualDesktopHostPoolCreateUpdate,
+		Delete: resourceArmVirtualDesktopHostPoolDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.VirtualDesktopHostPoolID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DevSpaceName(),
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"tags": tags.Schema(),
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Personal",
+					"Shared",
+				}, false),
+			},
+
+			"friendly_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 512),
+			},
+
+			"validation_environment": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"load_balancer_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"BreadthFirst",
+					"DepthFirst",
+				}, false),
+			},
+
+			"personal_desktop_assignment_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Automatic",
+					"Direct",
+				}, false),
+			},
+
+			"max_session_limit": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(0, 999999),
+			},
+
+			"preferred_app_group_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Preferred App Group type to display",
+				ValidateFunc: validation.StringInSlice([]string{
+					"Desktop",
+					"None",
+					"RailApplications",
+				}, false),
+				Default: "Desktop",
+			},
+
+			"registration_info": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"expiration_time": {
```

Comment:
would this make more sense as
```suggestion
						"expiration_date": {
```